### ISSUE TITLE
Is there a test framework where I can put OBFL-to-PEF tests?

### ISSUE THREAD

**Issue by bertfrees** [joeha480/dotify#114]
such as these: https://github.com/daisy/pipeline-mod-braille/blob/master/pipeline-braille-utils/dotify/dotify-formatter/src/test/xprocspec/test_obfl-to-pef.xprocspec

The tests do not necessarily have to be OBFL-to-PEF, they could be OBFL-to-text if we want to remove braille translation from the picture.

The framework should probably be added to dotify.formatter.impl.

**Comment by joeha480** [comment]
At the moment you'd have to add them to the Dotify subproject. It does not work in dotify.formatter.impl, because the implementations are not dependencies there. Perhaps we should add them to the test dependencies and move the tests there. But not now, I think.

**Comment by bertfrees** [comment]
What about mocking up the dependencies? E.g. a simple bypass translator and an PagedMediaWriter that writes to a simple text file.

**Comment by joeha480** [comment]
That's a good idea, however I think it's going to require some changes to the code to get it to work. In the meantime, I added the implementations as test dependencies.

**Comment by joeha480** [comment]
@bertfrees Can we close this issue? You can add tests to dotify.formatter.impl

**Comment by bertfrees** [comment]
Before moving my tests, we should look at whether we can/need to move the space normalization that I do in dotify-formatter ([obfl-normalize-space.xsl](https://github.com/daisy/pipeline-mod-braille/blob/master/pipeline-braille-utils/dotify-utils/dotify-formatter/src/main/resources/xml/obfl-normalize-space.xsl)).

**Comment by joeha480** [comment]
Does it break your tests?

**Comment by bertfrees** [comment]
Probably. I can of course remove the spaces manually. But maybe that way we're just covering up a bug?

**Comment by joeha480** [comment]
Not if you make a test specifically for that bug. I think that's actually more clear. But we could also do both: Wait for me to fix the bug and make a test for it.

**Comment by bertfrees** [comment]
OK, I'll make a test. Note that I'm not sure it's actually is a bug, I can't tell by just looking at the OBFL specification.

**Comment by bertfrees** [comment]
@joeha480 I've converted the tests but they fail and I'm having trouble finding out what goes wrong. Can you help me out?

**Comment by bertfrees** [comment]
OK never mind I just overlooked something because I was in a hurry. I made all the tests pass now. There are a bunch that I had to ignore, so that needs to be sorted out. And of course some selection needs to be done and they need to be given proper descriptions. But the selection can happen later I think, what you want now is as much test data as possible, to have a solid base for regression testing.
